app/javascript/controllers/message_scroller_controller.js

```diff
@@ -0,0 +1,40 @@
+window.lastMessageControllerInstance = null
+
+import { Controller } from "@hotwired/stimulus"
+import debounce from "utils/debounce"
+
+export default class extends Controller {
+  static values = {
+    scrollDown: { type: Boolean, default: false },
+    onlyScrollDownIfNearBottom: { type: Boolean, default: false }
+  }
+
+  connect() {
+    if (window.lastMessageControllerInstance) {
+      window.lastMessageControllerInstance.disconnect()
+      window.lastMessageControllerInstance = this
+    }
```

Comment:
There will potentially be A LOT of these stimulus controller instances created. When it was only firing once, that wasn't a big deal. But now that we have an event listener I don't want them all going off so I added this global tracker to disconnect previous ones as a new one is added.

Reply:
Interesting idea! I think you need to move line 19 out of the conditional?